src/modules/csm/csm.py

```diff
@@ -293,6 +296,11 @@ def current_frame_range(self, blockstamp: BlockStamp) -> tuple[EpochNumber, Epoc
                 EpochNumber(converter.get_epoch_by_slot(l_ref_slot) + converter.frame_config.epochs_per_frame)
             )
 
+        if last_processing_ref_slot > r_ref_slot:
+            raise CSMError(f"{last_processing_ref_slot=} > {r_ref_slot=}")
```

Comment:
and here as well. it isn't legit as I understand

Reply:
Done.